core/src/main/java/tech/pegasys/web3signer/core/Eth1Runner.java

```diff
@@ -147,36 +160,76 @@ protected ArtifactSignerProvider createArtifactSignerProvider(
       final Vertx vertx, final MetricsSystem metricsSystem) {
     return new DefaultArtifactSignerProvider(
         () -> {
-          final AzureKeyVaultSignerFactory azureFactory = new AzureKeyVaultSignerFactory();
-          final HashicorpConnectionFactory hashicorpConnectionFactory =
-              new HashicorpConnectionFactory();
-          try (final InterlockKeyProvider interlockKeyProvider = new InterlockKeyProvider(vertx);
-              final YubiHsmOpaqueDataProvider yubiHsmOpaqueDataProvider =
-                  new YubiHsmOpaqueDataProvider()) {
-
-            final Secp256k1ArtifactSignerFactory ethSecpArtifactSignerFactory =
-                new Secp256k1ArtifactSignerFactory(
-                    hashicorpConnectionFactory,
-                    baseConfig.getKeyConfigPath(),
-                    azureFactory,
-                    interlockKeyProvider,
-                    yubiHsmOpaqueDataProvider,
-                    EthSecpArtifactSigner::new,
-                    true);
-
-            return new SignerLoader(baseConfig.keystoreParallelProcessingEnabled())
-                .load(
-                    baseConfig.getKeyConfigPath(),
-                    "yaml",
-                    new YamlSignerParser(
-                        List.of(ethSecpArtifactSignerFactory),
-                        YamlMapperFactory.createYamlMapper(
-                            baseConfig.getKeyStoreConfigFileMaxSize())))
-                .getValues();
-          }
+          final List<ArtifactSigner> signers = new ArrayList<>();
+          final AzureKeyVaultFactory azureKeyVaultFactory = new AzureKeyVaultFactory();
+          registerClose(azureKeyVaultFactory::close);
+          final AzureKeyVaultSignerFactory azureSignerFactory =
+              new AzureKeyVaultSignerFactory(azureKeyVaultFactory);
+
+          signers.addAll(loadSignersFromConfigFiles(vertx, azureKeyVaultFactory).getValues());
+          signers.addAll(bulkLoadSigners(azureKeyVaultFactory, azureSignerFactory).getValues());
+          return signers;
         });
   }
 
+  private MappedResults<ArtifactSigner> loadSignersFromConfigFiles(
+      final Vertx vertx, final AzureKeyVaultFactory azureKeyVaultFactory) {
+    final AzureKeyVaultSignerFactory azureSignerFactory =
+        new AzureKeyVaultSignerFactory(azureKeyVaultFactory);
```

Comment:
nit: We're constructing azureSignerFactory here as well as the level above

Reply:
done